### PR TITLE
Fix multi-gpu sampling

### PR DESCRIPTION
**Pull request by kdha0727**
Thanks for your great work! When running `sample_seq2seq.py` in multi nodes in distributed, because of same seed, duplicated samples were generated. this PR fixes it and makes multi-GPU decoding more efficient.

### CODE REVIEW

**Review by summmeer — COMMENTED** [review]
Thanks for your contribution, however, the logic of writing the generated results to file should be examined meticulously. The distributed launcher may write the file simultaneously and the order of lines may be mixed up. However, in our current MBR algothrim, each line from files of different seeds should be aligned rigorously.

**Comment by kdha0727** [comment]
`for i in range: if i== rank` ~ `dist.barrier()` makes file i/o sequential. I have tested many times about the order of writing output file, and my code works sequentially.
However, if you intended to run sampling by different seeds (by `transformer.set_seed`) per node, the script I modified seems not to reflect the intention of yours. I would like to ask if you intented to set seed different by node. (The default sampling code sets equal seed.)

**Review by summmeer — COMMENTED** [review]
Need a little modification for `model_emb`

File: sample_seq2seq.py [hunk 1]
```diff
@@ -65,14 +69,13 @@ def main():
     pytorch_total_params = sum(p.numel() for p in model.parameters())
     logger.log(f'### The parameter count is {pytorch_total_params}')
 
-    model.to(dist_util.dev())
-    model.eval()
+    model.eval().requires_grad_(False).to(dist_util.dev())
 
     tokenizer = load_tokenizer(args)
     model_emb, tokenizer = load_model_emb(args, tokenizer)
```
[the remark this answers is not in the document]
Comment (kdha0727) [reply sample_seq2seq.py]: then `get_weights` function is no longer needed, so I reflected it.

[the remark this answers is not in the document]
Comment (kdha0727) [reply sample_seq2seq.py]: `model_emb` and `model_emb_copy` (in original code) were used in loading data, and denoising.
After loading all data, `model_emb` is no longer needed. `model_emb.to(dist_util.dev())` makes using same embedding in denoising function available.

**Review by summmeer — APPROVED** [review]
Tested, thanks for your contribution
